example/extensions/lib_custom_op/README.md

```diff
@@ -56,87 +72,111 @@ For building a library containing your own custom operator, compose a C++ source
 
-            MXReturnValue initialize(int version)
+```c++
+    MXReturnValue initialize(int version)
+```
 
 * [parseAttrs](./gemm_lib.cc#L118):
     * This function specifies number of input and output tensors for the custom operator; also this is where a custom operator can validate the attributes (ie. options) specified by the user.
 
-            MXReturnValue parseAttrs(
-                std::map<std::string,
-                std::string> attrs,
-                int* num_in,
-                int* num_out)
+```c++
+    MXReturnValue parseAttrs(
+        std::map<std::string,
+        std::string> attrs,
+        int* num_in,
+        int* num_out)
+```
 
 
 * [inferType](./gemm_lib.cc#L124):
     * This function specifies how the custom operator infers output data types using input data types.
 
-            MXReturnValue inferType(
-                std::map<std::string, std::string> attrs,
-                std::vector<int> &intypes,
-                std::vector<int> &outtypes)
+```c++
+    MXReturnValue inferType(
+        std::map<std::string, std::string> attrs,
+        std::vector<int> &intypes,
+        std::vector<int> &outtypes)
+```
 
 * [inferShape](./gemm_lib.cc#L143):
     * This function specifies how the custom operator infers output tensor shape using input shape.
 
-            MXReturnValue inferShape(
-                std::map<std::string, std::string> attrs,
-                std::vector<std::vector<unsigned int>> &inshapes,
-                std::vector<std::vector<unsigned int>> &outshapes)
+```c++
+    MXReturnValue inferShape(
+        std::map<std::string, std::string> attrs,
+        std::vector<std::vector<unsigned int>> &inshapes,
+        std::vector<std::vector<unsigned int>> &outshapes)
+```
 
 * [forward](./gemm_lib.cc#L56):
     * This function specifies the computation of the forward pass of the operator.
 
-            MXReturnValue forward(
-                std::map<std::string, std::string> attrs,
-                std::vector<MXTensor> inputs,
-                std::vector<MXTensor> outputs,
-                OpResource res)
+```c++
+    MXReturnValue forward(
+        std::map<std::string, std::string> attrs,
+        std::vector<MXTensor> inputs,
+        std::vector<MXTensor> outputs,
+        OpResource res)
+```
 
 * [REGISTER_OP(my_op_name)](./gemm_lib.cc#L169):
-    * This macro registers the custom operator and its properties to MXNet NDArray and Symbol APIs by its name.
-
-            REGISTER_OP(my_op_name)
-            .setForward(forward)
-            .setParseAttrs(parseAttrs)
-            .setInferType(inferType)
-            .setInferShape(inferShape);
+    * This macro registers the custom operator and its properties to MXNet NDArray and Symbol APIs by its name. Note that for operator running on CPU, you need pass the name of the context `"cpu"` when registering forward or backward function.
```

Comment:
This macro registers the custom operator with MXNet. You can register additional functions with the operator too. Note that you register functions for each context for Forward and Backward, here we show an example for CPU.

Reply:
lets move this to the end